cloud/src/meta-service/meta_service.cpp

```diff
@@ -1008,8 +1009,15 @@ void MetaServiceImpl::commit_rowset(::google::protobuf::RpcController* controlle
         DCHECK(rowset_meta.tablet_schema().has_schema_version());
         DCHECK_GE(rowset_meta.tablet_schema().schema_version(), 0);
         rowset_meta.set_schema_version(rowset_meta.tablet_schema().schema_version());
-        auto schema_key = meta_schema_key(
-                {instance_id, rowset_meta.index_id(), rowset_meta.schema_version()});
+        std::string schema_key;
+        if (rowset_meta.has_variant_type_in_schema()) {
+            // encodes schema in a seperate kv, since variant schema is volatile
```

Comment:
It will use much more memory and disk to store meta for each rowset.

Reply:
its necessary to spend such cost